torch/onnx/symbolic_opset9.py

```diff
@@ -1911,3 +1911,20 @@ def multinomial(g, input, num_samples, replacement=False, generator=None):
     return g.op("Multinomial", log_input,
                 dtype_i=sym_help.cast_pytorch_to_onnx['Long'],
                 sample_size_i=num_samples)
+
+
+def meshgrid(g, tensor_list):
+    tensors = [t for t in sym_help._unpack_list(tensor_list)]
+    out = []
+    repeats = [t.type().sizes()[0] if t.isCompleteTensor() else 0 for t in tensors]
+    for i, t in enumerate(tensors):
+        if t.isCompleteTensor():
+            shape_i = torch.ones(len(tensors), dtype=torch.int64)
+            shape_i[i] = t.type().sizes()[0]
+            repeats_i = torch.tensor(repeats, dtype=torch.int64)
+            repeats_i[i] = 1
+            t_reshaped = _reshape_from_tensor(g, t, g.op("Constant", value_t=torch.LongTensor(shape_i)))
+            out.append(repeat(g, t_reshaped, repeats_i))
+        else:
+            _unimplemented("meshgrid", "Unknown input rank. Cannot compute meshgird dimensions.")
```

Comment:
all inputs to meshgrid are 1-d tensor, is the error message meant to be "Unknown input shape"?

Also, in fact this can work without shape being statically known. I believe you could construct it dynamically by calling 
```
tensors_shape = [g.op("Shape", t) for t in tensors]
out_shape = g.op("Concat", *tensors_shape, axis_i=0)
tensors_reshaped = [_reshape_from_tensor(g, t, out_shape) for t in tensors]
```

There is an edge case though, that for this op scalars (rank 0) is treated as rank 1. So to export for all cases we need to do a protective reshape as well...
```
tensors = [view(g, t, torch.LongTensor([-1])) for t in tensors]
```

And add the scalar test case.

```
        class Meshgrid(torch.nn.Module):
            def forward(self, x, y, z):
                return torch.meshgrid(x, y, z)

        x = torch.ones(3, requires_grad=True)
        y = torch.zeros(4, requires_grad=True)
        z = torch.tensor(2.0)
        self.run_test(Meshgrid(), (x, y, z))
```